json-rpc/types/src/views.rs

```diff
@@ -33,14 +31,49 @@ pub struct AmountView {
 }
 
 impl AmountView {
-    fn new(amount: u64, currency: &IdentStr) -> Self {
+    fn new(amount: u64, currency: &str) -> Self {
         Self {
             amount,
             currency: currency.to_string(),
         }
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub enum AccountRoleView {
+    #[serde(rename = "unhosted")]
+    Unhosted,
+    #[serde(rename = "empty")]
+    Empty,
+    #[serde(rename = "child_vasp")]
+    ChildVASP { parent_vasp_address: BytesView },
+    #[serde(rename = "parent_vasp")]
+    ParentVASP {
+        human_name: String,
+        base_url: String,
+        expiration_time: u64,
+        compliance_key: BytesView,
+    },
+}
+
+impl AccountRoleView {
+    pub fn new(role: AccountRole) -> Self {
```

Comment:
pls  follow existing convention to. implement std::convert::From  trait

and it is best to be future proof to have an "Unknown" type int he enum